homepage/models.py

```diff
@@ -61,6 +72,17 @@ class FollowedUserCourses(models.Model):
     def __str__(self):
         return f'user = {self.user}, course = {self.course}'
 
+    @staticmethod
+    def get_courses_followed_by_appUser(appUser):
+        allFollowedUserCourses = FollowedUserCourses.objects.all()
+        courses_followed_by_appUser = list()
+
+        for user_course_pair in allFollowedUserCourses:
+            if (user_course_pair.user == appUser):
+                courses_followed_by_appUser.append(user_course_pair.course)
+

```

Comment:
As Ori said in the previous PR - please consider using `SomeModel.objects.filter(...)` as it returns a QuerySet and not a list, which seems a more correct way to address a query

Reply:
The only option in this case is `FollowedUserCourses.objects.filter(user=appUser)`
which unfortunately does not help. 

This command will return a QuerySet which consist of 'FollowedUserCourses' elements.
Then i still need to iterate other all of the elements in the QuerySet to get only the 'Course' element

Reply:
I did some tests to implement it with objects.filter. 
Let me know if this is clear or if you have ideas for improvements:

![image](https://user-images.githubusercontent.com/79373827/116124777-6781d300-a6cd-11eb-8d3a-52719045943c.png)

Reply:
@LeonRabinovich 
did you consider using this : `list(FollowedUserCourses.filter(user=user).values_list('course'))` to turn the QuerySet into a list?
For example, in my Course Model, I used this static method:
```
    @staticmethod
    def test():
        return list(Course.objects.filter(course_id__contains=1).values_list('name'))
```
and received this output in the shell:
```
>>> res = Course.test()
>>> print(res)
[('Resonance in Runes and Signs',), ('Grammatica in Arithmancy',), ('UnFogging the Future',), ('Numerology',)]
```

Reply:
@VeraTs Yes I think I'll use it
The problem began when I thought that maybe I should return a QuerySet instead of the Courses/Pro's.
In that case if I can return a list of objects (which I think is easier to work with for the view) then your code is perfect.

Reply:
@ellaml  @LeonRabinovich 
The methods I wrote about above give back a list of values, not objects.
If it's a list of objects, then a QuerySet is the best way to go, really. 
For example:
for this function: 
```
    @staticmethod
    def test():
        return Prerequisites.objects.filter(course_id=12357)
```

I get this output:
`<QuerySet [<Prerequisites: req_cour = UnFogging the Future, desired_cour = Numerology, req_code = 1>]>`

and can handle it like so:
```
>>> res[0].course_id
<Course: Numerology>
>>> res[0].course_id.print_details()
```

and print_details() works.

Reply:
> @LeonRabinovich you say that you still have to iterate over the elements in the QuerySet, but let me explain the difference.
> Let say you have 2000 rows in this table, when you write `allFollowedUserCourses = FollowedUserCourses.objects.all()`
> you're going to retrieve all of them, and iterate over **2000 objects**.
> If the user has only 5 followed courses, and you call `FollowedUserCourses.objects.filter(user=appUser)`, then you have to iterate over **5 objects**.

Yes i agree Ori, however the objects.filter() method probably has to iterate all of the 2000 objects anyways in the background to filter the required objects .. 
it is probably more efficient than a my own iteration though, which is why I did change the method to use filter()

Reply:
@VeraTs 

Thanks for you help,
I will do some more tests and see if i can write it more efficiently

When I'm done I will commit with all of the changes discussed in the PR

Reply:
@VeraTs 
I had no luck using values_list.
even though I specify ('course') as the parameter for values_list - i still get IDs instead of objects.

I decided to keep the previous code (as i sent in the picture).
This will be updated in the new PR.

![image](https://user-images.githubusercontent.com/79373827/116552366-9da79880-a901-11eb-989a-d9a695bd8b65.png)